lib/v2/screens/profile_screens/guardians/interactor/usecases/get_guardians_usecase.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:seeds/v2/datasource/local/settings_storage.dart';
+import 'package:seeds/v2/datasource/remote/api/members_repository.dart';
+import 'package:seeds/v2/datasource/remote/firebase/firebase_database_guardians_repository.dart';
+import 'package:seeds/v2/datasource/remote/model/firebase_models/guardian_model.dart';
+import 'package:seeds/v2/datasource/remote/model/member_model.dart';
+
+class GetGuardiansUseCase {
+  final FirebaseDatabaseGuardiansRepository _repository = FirebaseDatabaseGuardiansRepository();
+  final MembersRepository _membersRepository = MembersRepository();
+
+  Stream<List<GuardianModel>> getGuardians() {
+    return _repository
+        .getGuardiansForUser(settingsStorage.accountName)
+        .asyncMap((List<GuardianModel> event) => getMemberData(event));
+  }
+
+  Future<List<GuardianModel>> getMemberData(List<GuardianModel> guardians) async {
```

Comment:
First we get the guardians from firebase, 
Then we get their data from chain. 

return a stream of that